### PR TITLE
Indicate AKS Engine as stable, CAPZ for new features

### PR DESCRIPTION
**Pull request by craiglpeters**
**Reason for Change**:
README update to clarify aks-engine and cluster api

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [X] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [X] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by devigned — COMMENTED** [review]
What would you think about adding a `## Project status` blurb?

**Comment by craiglpeters** [comment]
@devigned You mean to have a section specifically called out? I'd like to ask @jadarsie and @Michael-Sinz what they think

File: README.md [hunk 1]
```diff
@@ -5,11 +5,11 @@
 [![GoDoc](https://godoc.org/github.com/Azure/aks-engine?status.svg)](https://godoc.org/github.com/Azure/aks-engine)
 [![Go Report Card](https://goreportcard.com/badge/github.com/Azure/aks-engine)](https://goreportcard.com/report/github.com/Azure/aks-engine)
 
-AKS Engine is the easiest way to provision a self-managed Kubernetes cluster on Azure.
+AKS Engine is a legacy ARM template-driven way to provision a self-managed Kubernetes cluster on Azure.
```
Comment (Michael-Sinz) [inline README.md:8]: I would phrase it as "AKS Engine is an ARM template-driven way"

I would not make a statement that ARM templates are "legacy" - aks-engine may become legacy but ARM templates have not (yet?) been put onto the legacy pile.  (They may not be the best way to deal with kubernetes clusters but that does not make them legacy yet)

File: README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ Please see the [FAQ][] for answers about AKS Engine and its progenitor ACS-Engin
 
 ## Join the community
 
-Want to get involved? The [community guide][community] covers everything you need to know about the AKS Engine community and how you can contribute. The [developer guide][developer-guide] will help you onboard as a developer.
+If you are committed to using AKS Engine longer term and would like to become a project maintainer, please reach out to us via the [#aks-engine-dev Slack channel](https://kubernetes.slack.com/archives/CU1CXUHN0)! The [community guide][community] covers everything you need to know about the AKS Engine community and how you can contribute. The [developer guide][developer-guide] will help you onboard as a developer. The AKS Engine community is committed to integrating and validating new versions of Kubernetes into AKS Engine. We encourage AKS Engine users to evaluate moving to CAPZ as it provides stronger support for managing the cluster lifecycle compared to AKS Engine.
```
Comment (Michael-Sinz) [inline README.md:26]: Is CAPZ ready for production work?  (At least at the level of aks-engine?)  I would be careful how we phrase the push to CAPZ.  I really want people to look at and play with CAPZ but are we ready for major customers to make production clusters that way?  (Just asking - I would love it if that were the case)

**Comment by ritazh** [comment]
Thank you for opening this PR @craiglpeters! Can we also explicitly state that the project is going into maintenance mode and will not be adding new features?

**Comment by craiglpeters** [comment]
@ritazh no. That is up to the community if it takes new features or not. What is changing is that our team isn't going to be driving new features

File: README.md (hunk 1 above)
Comment (craiglpeters) [reply README.md:8]: Good point. Will reword - happy to take suggestions

File: README.md (hunk 2 above)
Comment (craiglpeters) [reply README.md:26]: CAPZ is ready for production work. Does it (or will it ever have) all the features of aks-engine? No, it likely will not

File: README.md (hunk 1 above)
Comment (jackfrancis) [reply README.md:8]: "AKS Engine is a legacy tool that leverages ARM templates to..."

**Comment by jadarsie** [comment]
> @devigned You mean to have a section specifically called out? I'd like to ask @jadarsie and @Michael-Sinz what they think

Thanks Craig, my initial thoughts were already brought up, looping @walterov

**Comment by walterov** [comment]
@craiglpeters, as you know AKS Engine continues to be the technology for managing K8s in Azure Stack Hub. It will continue to be the case for over a year at least while we ramp up AKS RP in ASH and customers migrate to it. So we continue enabling features in the tool, although these features are new to ASH, they are not to AKS engine in Azure, but they do take code changes. So I have two concerns, 1. Customers read your comments and think that AKS Engine is deprecated everywhere, which is not the case and their decision to acquire ASH is then unnecessarily hampered. 2. They try to use CAPZ on ASH and naturally run into issues.
Could we change your language a bit to make sure customers understand that AKS engine continues to be maintained and enhanced for ASH customers as we bring more AKS features to ASH?

**Comment by craiglpeters** [comment]
@jadarsie, @walterov, @Michael-Sinz, @devigned, and @ritazh - I'm going to scrap this one, take the feedback, and start over.

**Comment by craiglpeters** [comment]
Please take a look at https://github.com/Azure/aks-engine/pull/4312

**Pull request closed without merging**